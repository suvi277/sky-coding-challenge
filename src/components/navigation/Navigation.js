import React from 'react';
import { ROUTES } from 'routes/routeConfig';
import { NavLink } from 'react-router-dom';
import './Navigation.scss';

const Navigation = ({ routes }) => {
	const getNavList = (route, i) => {
		let navItems = '';

		if (route.key !== 'notFound') {
			navItems = (
				<li key={i} className="nav-item">
					<NavLink className="nav-link p-3" to={route.path} exact={route.exact}>
						{route.label}
					</NavLink>
				</li>
			);
		}
		return navItems;
	};
	return (
		<ul className="nav justify-content-end">
			{ROUTES.map((route, i) => {
				return getNavList(route, i);
			})}
		</ul>
	);
};

export default Navigation;
