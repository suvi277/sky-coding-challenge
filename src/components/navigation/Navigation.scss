@import "../../_shared/styles/variable";

.nav-link {
	&.active {
		background-color: $primary;
        color: $light;
    }
    
    &:not(.active):hover {
        background-color: $secondary;
		color: $light;
    }
}
