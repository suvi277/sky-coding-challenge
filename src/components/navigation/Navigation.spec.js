import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';
import { NavLink } from 'react-router-dom';

describe('Navigation Component', () => {
	const mockRouteConfig = [
		{
			path: '/',
			key: 'home',
			label: 'Home',
			exact: true,
			component: () => {}
		},
		{
			path: '/content',
			key: 'streams',
			label: 'Streams',
			component: () => {}
		}
	];
	const wrapper = shallow(<Navigation routes={mockRouteConfig} />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should exist NavLink Component', () => {
		expect(wrapper.exists(NavLink)).toBeTruthy();
	});
	it('should render two nav-item tag', () => {
		expect(wrapper.find('.nav-item')).toHaveLength(2);
	});
});
