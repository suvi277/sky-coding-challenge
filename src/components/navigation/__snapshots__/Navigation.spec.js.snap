// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders as expected 1`] = `
<ul
  className="nav justify-content-end"
>
  <li
    className="nav-item"
    key="0"
  >
    <NavLink
      className="nav-link p-3"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
  </li>
  <li
    className="nav-item"
    key="1"
  >
    <NavLink
      className="nav-link p-3"
      to="/content"
    >
      Streams
    </NavLink>
  </li>
</ul>
`;
